### PR TITLE
[libheif] x265 as feature to allow non-GPL build

### PR DESCRIPTION
**Pull request by MajorMurphy**
x265 is still included as a default feature when installed as:

`vcpkg.exe install libheif`

 but to avoid linking to a GPL library, x265 can now be optionally excluded with:

`vcpkg.exe install libheif[core,libde265]`

 (libheif cannot encode without x265, only decode)

### Checklist
(please forgive if completed inaccurately)

- [✔️] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [N/A ] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [N/A] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [ N/A] The versioning scheme in `vcpkg.json` matches what upstream says.
- [N/A] The license declaration in `vcpkg.json` matches what upstream says.
- [ N/A] The installed as the "copyright" file matches what upstream says.
- [ N/A] The source code of the component installed comes from an authoritative source.
- [ N/A ?] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [ N/A] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [ N/A] Only one version is in the new port's versions file.
- [ N/A] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by MajorMurphy** [comment]
@microsoft-github-policy-service agree

**Comment by MajorMurphy** [comment]
Briefly looking at the x86_windows build log, the failure seems to be due to re-using the same version number.   
The version specified in portfile.json shouldn't be incremented when libheif's actual version number remains unchanged, correct?

**Comment by LilyWangLL** [comment]
Please update the port's version fields and then run:
```
vcpkg x-add-version libheif
git add versions
git commit -m "Update version database"
to add the new version.
```

File: ports/libheif/vcpkg.json [hunk 1]
```diff
@@ -10,15 +11,31 @@
...
+        "libde265"
+      ]
+    },
+    "x265": {
+      "description": "HEVC encoding via x265",
+      "dependencies": [
+        "x265"
+      ]
+    }
+  }
```
Comment (dg0yt) [inline ports/libheif/vcpkg.json:40]: I think this should be a single feature `hevc`, modeling the capability, not mirroring the dependencies.

And IIRC it is now also possible to have a license field on individual features.

**Comment by MajorMurphy** [comment]
@dg0yt `hevc` is now the single feature.  
x265 can be avoided by building as: `vcpkg install libheif[core]`
GPL license is noted as a field under the feature.

@LilyWangLL thanks for the assistance with the versioning

File: ports/libheif/portfile.cmake [hunk 2]
```diff
@@ -7,13 +7,21 @@ vcpkg_from_github(
...
+	"-DWITH_EXAMPLES=OFF"
+	"-DWITH_DAV1D=OFF"
+	"-DPKG_CONFIG_USE_CMAKE_PREFIX_PATH=ON"
+	)
+	
+if("hevc" IN_LIST FEATURES)
+list(APPEND OPTIONS "-DWITH_X265=ON")
+else()
+list(APPEND OPTIONS "-DWITH_X265=OFF")
+endif()
```
Comment (LilyWangLL) [inline ports/libheif/portfile.cmake:20]:
```suggestion
	
vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
    hevc    WITH_X265
)
```

File: ports/libheif/portfile.cmake [hunk 3]
```diff
@@ -7,13 +7,21 @@ vcpkg_from_github(
...
 
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
-    OPTIONS
-        -DWITH_EXAMPLES=OFF
-        -DWITH_DAV1D=OFF
-        -DPKG_CONFIG_USE_CMAKE_PREFIX_PATH=ON
+    OPTIONS ${OPTIONS}
```
Comment (LilyWangLL) [inline ports/libheif/portfile.cmake:24]:
```suggestion
    OPTIONS
        -DWITH_EXAMPLES=OFF
        -DWITH_DAV1D=OFF
        -DPKG_CONFIG_USE_CMAKE_PREFIX_PATH=ON
        ${FEATURE_OPTIONS}
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/libheif/portfile.cmake (hunk 2 above)
Comment (MajorMurphy) [reply ports/libheif/portfile.cmake]: I added a FEATURE keyword as suggested by a deprecation warning.
`vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
	FEATURES
		hevc    WITH_X265
)`

**Review by LilyWangLL — APPROVED** [review]

**Pull request merged**